OEP-6/OEP-6.mediawiki

```diff
@@ -0,0 +1,394 @@
+
+Returns the <code>Signature</code> object containing both the <code>publicKey</code> and signature <code>data</code> needed for signature validation.
+
+This method does not require the dApp to disclose the message itself, only the hash. 
+
+Because malicious dApp can hash a real prepared transfer transaction and plant it for signing, that posses a risk to the user.
+Therefore the hash is prepended with known string '''Ontology message:''', and only this hash is put for signing.
+
+* Rejects with '''NO_ADDRESS''' in case the user is not signed in or has no account or identity
+* Rejects with '''MALFORMED_MESSAGE''' in case the <code>message</code> is not hex encoded
+
+====verifyMessageHash====
+<pre>
+function verifyMessageHash({ messageHash: string, signature: Signature }): Promise<boolean>
+</pre>
+
+Verifies that the <code>signature</code> is valid signature of <code>messageHash</code>. 
+
+This method does not require the dApp to disclose the message itself, only the hash. The hash is prepended with known string '''Ontology message:''' and only this hash is put for verification.
+
+* Rejects with '''MALFORMED_MESSAGE''' in case the <code>message</code> is not hex encoded
+* Rejects with '''MALFORMED_SIGNATURE''' in case the <code>signature</code> is not in valid format
+
+====signMessage====
+<pre>
+function signMessage({ message: string }): Promise<Signature>
+</pre>
+
+Initiates signing of arbitrary <code>message</code> by the user account or identity. dAPI provider should provide the user with a way for selecting which account or identity will sign the message.
+
+Returns the <code>Signature</code> object containing both the <code>publicKey</code> and signature <code>data</code> needed for signature validation.
+
+This method provide the dAPI provider with the message body, which it should display to the user prior to signing (e.g.: a contract to sign).
+
+* Rejects with '''NO_ADDRESS''' in case the user is not signed in or has no account or identity
+
+====verifyMessage====
+<pre>
+function verifyMessage({ message: string, signature: Signature }): Promise<boolean>
+</pre>
+
+Verifies that the <code>signature</code> is valid signature of <code>message</code>. 
+
+* Rejects with '''MALFORMED_MESSAGE''' in case the <code>message</code> is not hex encoded
+* Rejects with '''MALFORMED_SIGNATURE''' in case the <code>signature</code> is not in valid format
+
+
+===SmartContract===
+The main functionality of this component is invocation of smart contract methods.
+
+====invoke====
+<pre>
+type ParameterType = 'Boolean' | 'Integer' | 'ByteArray' | 'Struct' | 'Map' | 'String';
+
+interface Parameter {
+    type: ParameterType;
+    value: any;
+}
+
+type Result = string[];
+
+interface Response {
+    transaction: string;
+    results: Result[];
+}
+
+function invoke({ contract: string, method: string, parameters?: Parameter[], 
```

Comment:
Can we please update all references to a contract script hash from `contract`, to `scriptHash` to have a more specific description of this argument?

Reply:
I agree.